### ISSUE TITLE
react native 0.76 upgrade - ninja build fails

### ISSUE THREAD

**Issue by IncrediblePony** [th3rdwave/react-native-safe-area-context#541]
```plaintext
  In file included from /workspace/my_project/node_modules/react-native-safe-area-context/android/src/main/jni/../../../build/generated/source/codegen/jni/react/renderer/components/safeareacontext/ShadowNodes.h:13:
  /workspace/my_project/node_modules/react-native-safe-area-context/android/src/main/jni/../../../build/generated/source/codegen/jni/react/renderer/components/safeareacontext/EventEmitters.h:12:10: fatal error: 'react/renderer/components/view/ViewEventEmitter.h' file not found
  #include <react/renderer/components/view/ViewEventEmitter.h>
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.
  [72/125] Building CXX object rnreanimated_autolinked_build/CMakeFiles/react_codegen_rnreanimated.dir/react/renderer/components/rnreanimated/EventEmitters.cpp.o
  [73/125] Building CXX object rnpermissions_autolinked_build/CMakeFiles/react_codegen_rnpermissions.dir/react/renderer/components/rnpermissions/ShadowNodes.cpp.o
  [74/125] Building CXX object rnpermissions_autolinked_build/CMakeFiles/react_codegen_rnpermissions.dir/react/renderer/components/rnpermissions/rnpermissionsJSI-generated.cpp.o
  [75/125] Building CXX object rnpermissions_autolinked_build/CMakeFiles/react_codegen_rnpermissions.dir/react/renderer/components/rnpermissions/Props.cpp.o
  [76/125] Building CXX object rnreanimated_autolinked_build/CMakeFiles/react_codegen_rnreanimated.dir/react/renderer/components/rnreanimated/States.cpp.o
  [77/125] Building CXX object rnreanimated_autolinked_build/CMakeFiles/react_codegen_rnreanimated.dir/react/renderer/components/rnreanimated/ComponentDescriptors.cpp.o
  [78/125] Building CXX object rnpermissions_autolinked_build/CMakeFiles/react_codegen_rnpermissions.dir/rnpermissions-generated.cpp.o
  [79/125] Building CXX object rnreanimated_autolinked_build/CMakeFiles/react_codegen_rnreanimated.dir/react/renderer/components/rnreanimated/ShadowNodes.cpp.o
  [80/125] Building CXX object rnreanimated_autolinked_build/CMakeFiles/react_codegen_rnreanimated.dir/react/renderer/components/rnreanimated/Props.cpp.o
  [81/125] Building CXX object rnreanimated_autolinked_build/CMakeFiles/react_codegen_rnreanimated.dir/rnreanimated-generated.cpp.o
  [82/125] Building CXX object rnreanimated_autolinked_build/CMakeFiles/react_codegen_rnreanimated.dir/react/renderer/components/rnreanimated/rnreanimatedJSI-generated.cpp.o
  [83/125] Building CXX object CMakeFiles/appmodules.dir/workspace/my_project/android/app/build/generated/autolinking/src/main/jni/autolinking.cpp.o
  ninja: build stopped: subcommand failed.
  
  C++ build system [build] failed while executing:
      /Android/Sdk/cmake/3.22.1/bin/ninja \
        -C \
        /workspace/my_project/android/app/.cxx/Debug/1421s6d5/arm64-v8a \
        appmodules \
        react_codegen_rnscreens \
        react_codegen_rnsvg \
        react_codegen_safeareacontext
    from /workspace/my_project/android/app
```

My OS: Ubuntu 22.04 LTS
Platform: Android

`package.json`
```json
{
  "dependencies": {
    "@apollo/client": "3.8.4",
    "@formatjs/intl-datetimeformat": "6.10.3",
    "@formatjs/intl-displaynames": "6.5.2",
    "@formatjs/intl-getcanonicallocales": "2.2.1",
    "@formatjs/intl-listformat": "7.4.2",
    "@formatjs/intl-locale": "3.3.4",
    "@formatjs/intl-numberformat": "8.7.2",
    "@formatjs/intl-pluralrules": "5.2.6",
    "@formatjs/intl-relativetimeformat": "11.2.6",
    "@gorhom/bottom-sheet": "^4.6.3",
    "@invertase/react-native-apple-authentication": "^2.3.0",
    "@miblanchard/react-native-slider": "^2.3.1",
    "@react-native-community/geolocation": "3.4.0",
    "@react-native-community/netinfo": "^11.3.1",
    "@react-native-firebase/app": "17.3.2",
    "@react-native-firebase/messaging": "17.3.2",
    "@react-native-google-signin/google-signin": "^12.1.0",
    "@react-navigation/drawer": "6.6.6",
    "@react-navigation/native": "^6.0.8",
    "@react-navigation/native-stack": "6.9.13",
    "@reduxjs/toolkit": "1.9.5",
    "@sentry/react-native": "5.32.0",
    "@shopify/flash-list": "^1.6.4",
    "apollo-link-queue": "3.1.0",
    "apollo-link-serialize": "4.0.0",
    "apollo-upload-client": "17.0.0",
    "app-modules": "file:app-modules",
    "autolinker": "^4.0.0",
    "firebase": "9.23.0",
    "flag-icons": "6.11.1",
    "fuse.js": "6.6.2",
    "global": "^4.4.0",
    "graphql": "16.8.1",
    "intl-pluralrules": "2.0.1",
    "lodash": "4.17.21",
    "parse5": "7.1.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-native": "^0.76.0",
    "react-native-blob-util": "0.19.1",
    "react-native-branch": "^6.2.2",
    "react-native-circular-progress": "1.3.9",
    "react-native-confirmation-code-field": "^7.4.0",
    "react-native-country-picker-modal": "2.0.0",
    "react-native-date-picker": "^5.0.7",
    "react-native-device-info": "10.11.0",
    "react-native-draggable-flatlist": "^4.0.1",
    "react-native-fbsdk-next": "^12.1.1",
    "react-native-fs": "^2.18.0",
    "react-native-gesture-handler": "^2.20.2",
    "react-native-get-random-values": "^1.9.0",
    "react-native-google-places-autocomplete": "2.5.1",
    "react-native-haptic-feedback": "^2.3.1",
    "react-native-hyperlink": "0.0.22",
    "react-native-image-picker": "7.1.2",
    "react-native-inappbrowser-reborn": "3.7.0",
    "react-native-linear-gradient": "2.8.3",
    "react-native-localize": "2.2.4",
    "react-native-map-clustering": "3.4.2",
    "react-native-maps": "2.0.0-beta.14",
    "react-native-orientation-locker": "^1.7.0",
    "react-native-paper-dates": "0.19.7",
    "react-native-permissions": "3.9.2",
    "react-native-reanimated": "3.16.1",
    "react-native-render-html": "^6.3.4",
    "react-native-safe-area-context": "^4.11.1",
    "react-native-screens": "^3.34.0",
    "react-native-svg": "^15.8.0",
    "react-native-svg-charts": "^5.4.0",
    "react-native-svg-transformer": "^1.5.0",
    "react-native-watch-connectivity": "^1.1.0",
    "react-native-webview": "^13.10.5",
    "react-redux": "8.1.2",
    "redux": "4.2.1",
    "redux-persist": "6.0.0",
    "redux-persist-filesystem-storage": "4.2.0",
    "redux-saga": "1.2.3",
    "reselect": "^4.1.8",
    "standard-version": "9.5.0",
    "uuid": "^9.0.1",
    "validator": "13.11.0"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2",
    "@babel/preset-env": "^7.25.3",
    "@babel/runtime": "^7.25.0",
    "@react-native-community/cli": "15.0.0-alpha.2",
    "@react-native-community/cli-platform-android": "15.0.0-alpha.2",
    "@react-native-community/cli-platform-ios": "15.0.0-alpha.2",
    "@react-native/babel-preset": "0.76.0",
    "@react-native/metro-config": "0.76.0",
    "@react-native/typescript-config": "0.76.0",
    "@testing-library/react-native": "^12.4.3",
    "@tsconfig/react-native": "^3.0.0",
    "@types/apollo-upload-client": "^17.0.4",
    "@types/jest": "^29.5.11",
    "@types/lodash": "^4.14.199",
    "@types/react": "^18.2.6",
    "@types/react-native-svg-charts": "^5.0.15",
    "@types/react-native-vector-icons": "6.4.14",
    "@types/react-test-renderer": "^18.0.0",
    "@types/uuid": "9.0.4",
    "@typescript-eslint/eslint-plugin": "^6.7.2",
    "@typescript-eslint/parser": "^6.8.0",
    "babel-jest": "^29.6.3",
    "babel-plugin-inline-import": "^3.0.0",
    "eslint": "^8.50.0",
    "eslint-config-glfr": "git+ssh://git@github.com:IngeniumGolf/eslint-config-glfr.git#master",
    "eslint-config-prettier": "^9.0.0",
    "eslint-import-resolver-typescript": "^3.6.1",
    "eslint-plugin-import": "^2.28.1",
    "eslint-plugin-react": "^7.33.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "expo": "^49.0.13",
    "husky": "8.0.3",
    "jest": "^29.2.1",
    "npm": "^10.2.0",
    "prettier": "3.0.3",
    "react-native-asset": "2.1.1",
    "react-native-vector-icons": "10.1.0",
    "react-test-renderer": "^18.2.0",
    "typescript": "^5.4.5"
  },
  "engines": {
    "node": ">=18"
  },
  "packageManager": "yarn@3.6.4",
  "name": "my_project",
  "private": true,
  "reactNativePermissionsIOS": [
    "Camera",
    "LocationAccuracy",
    "LocationAlways",
    "LocationWhenInUse",
    "Notifications",
    "PhotoLibrary"
  ],
  "scripts": {
    "android": "react-native run-android",
    "bump": ".scripts/bump",
    "fix": ".scripts/fix src/ util/ app-modules/ !node_modules/ .eslintrc.js package.json",
    "ios": "react-native run-ios --simulator='iPhone 15 Plus'",
    "ios-permissions": "react-native setup-ios-permissions && pod-install",
    "lint": ".scripts/lint src/ util/",
    "postinstall": "husky install",
    "start": "react-native start",
    "rn-asset": "npx react-native-asset",
    "test": "jest --verbose"
  },
  "version": "5.2.1",
  "versionBuild": "01"
}
```

**Comment by hasithmeth** [comment]
Facing the same problem on Macbook M1
Platform: Android

**Comment by hasithmeth** [comment]
@IncrediblePony you can follow these instructions and get your build working until the fix is pulled.
https://github.com/th3rdwave/react-native-safe-area-context/issues/539#issuecomment-2436474432

**Comment by zoobibackups** [comment]
samme issu on a fresh projet.

**Comment by tuan0919** [comment]
same issue with fresh project

**Comment by khushal87** [comment]
+1

**Comment by mertcancetinok** [comment]
+1

**Comment by JorgeDanilo** [comment]
same problem, i'm using android

**Comment by janicduplessis** [comment]
I just released a new version that should support 0.76, can you confirm it now works?

**Comment by JorgeDanilo** [comment]
> I just released a new version that should support 0.76, can you confirm it now works?

It worked for me, thanks
